### PR TITLE
General improvements 3

### PR DESCRIPTION
**Pull request by jgriffiths**
Misc stuff. Feedback on  	5f2fb80 would be good.

The component makefiles will need to be gradually cleaned up to allow us to provide proper targets to/expose full dependencies from the top level Makefile. Patches for those will come in due course.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -179,16 +179,13 @@ $(get-sources-tgt): build-info
...
-	@if ! which rpm >/dev/null 2>&1; then
-		echo "WARNING: rpm executable not found (are you on cygwin?)"; \
-	elif [ $(VERBOSE) -gt 0 ]; then \
-		echo "currently installed dependencies:"; \
-		rpm -q $(DEPENDENCIES) || exit 1; \
-	else \
-		rpm -q $(DEPENDENCIES) >/dev/null 2>&1 || { echo "ERROR: call 'make install-deps' to install missing dependencies"; exit 1; }; \
-	fi
+.PHONY: check.rpm check-depend
+check.rpm: $(if $(shell which rpm 2>/dev/null), /bin/true, please.install.rpm.and.try.again);
```
Comment (marmarek) [inline Makefile]: Any idea how to support also Debian host here? It would use `dpkg -l` instead of `rpm -q`.

File: Makefile [hunk 2]
```diff
@@ -172,7 +174,7 @@ help:
 	@echo "to operate on subset of components. Example: make COMPONENTS=\"gui\" get-sources"
 
 
-get-sources-sort = $(BUILDER_PLUGINS) $(filter-out builder $(BUILDER_PLUGINS), $(COMPONENTS))
+get-sources-sort = $(BUILDER_PLUGINS) $(filter-out $(BUILDER_PLUGINS), $(COMPONENTS_NO_BUILDER))
```
Comment (marmarek) [inline Makefile]: Actually I think `builder` shouldn't be excluded here. See QubesOS/qubes-issues#1405. But that worth a separate commit :)

Comment (jgriffiths) [inline Makefile]: OK I will have a look at making that work.

**Comment by jgriffiths** [comment]
>Any idea how to support also Debian host here?

Yes its easy enough, but it gets complicated by some needless complication in this target:

- We don't need to check for rpm (or dpkg) really. How can you install the O/S without them? This could only be useful on non-fedora/non-debian systems like windows where I would hope you wouldn't make this target anyway.

-The verbosity levels for one target is a bit silly. Either the deps are OK and the command should be silent, or they aren't and it should complain.

I'll push a simpler version that should handle debian too, I cant test it right now though (no debian VM installed).

**Comment by jgriffiths** [comment]
So 2c66620 would do the job provided that a) the dependencies are named the same and 2) dpkg -I returns an error code when querying an uninstalled package.

However, this doesn't really help us given that the dependencies are rpm related (rpm-sign etc) and install-deps calls yum anyway.

If you would like to raise a bug for the build system as a whole it might be good to talk about how to generalise these things going forward. With the current number of platforms and platform versions the complexity is really starting to show.

**Pull request merged**